### PR TITLE
feat(discards): Add discarded hash filtering stats to tsdb

### PR DESCRIPTION
**Pull request by macqueen**
This is setting the `start_time` earlier in ingestion, so not totally sure if that is going to have adverse consequences. The other thing I could do is have some separate `tsdb_time` arg, but that also seems gross.

This also passes an explicit timestamp to existing tsdb.incr calls.

### CODE REVIEW

File: src/sentry/web/api.py [hunk 1]
```diff
@@ -353,7 +357,8 @@ def process(self, request, project, key, auth, helper, data, **kwargs):
             project=project,
             sender=type(self),
         )
-
+        start_time = time()
+        tsdb_start_time = datetime.datetime.fromtimestamp(start_time).replace(tzinfo=pytz.utc)
```
Comment (tkaemming) [inline src/sentry/web/api.py]: Use `sentry.utils.dates.to_timestamp` here.

**Review by tkaemming — CHANGES_REQUESTED** [review]
Seems good to me other than the issue noted inline

File: src/sentry/web/api.py (hunk 1 above)
Comment (macqueen) [reply src/sentry/web/api.py]: hmm do you mean `sentry.utils.dates.to_datetime`?

Comment (tkaemming) [reply src/sentry/web/api.py]: 🙃  Yep

**Review by tkaemming — APPROVED** [review]
Beep boop looks good

**Pull request merged**